heron/api/src/java/org/apache/heron/api/metric/CountStatAndMetric.java

```diff
@@ -176,7 +176,7 @@ private synchronized void rotate(long value, long timeSpent,
   }
 
   synchronized Map<String, Long> getTimeCounts(long now) {
-    Map<String, Long> ret = new HashMap<>();
+    Map<String, Long> ret = new ConcurrentHashMap<>();
```

Comment:
ConcurrentHashMap  is not needed here I think. This is a get function and it is already synchronized.